ui/src/pages/listing/New.tsx

```diff
@@ -26,44 +23,51 @@ export const New : React.FC = () => {
   const [ quotedInstrumentLabel, setQuotedAssetLabel ] = useState("");
   const [ id, setId ] = useState("");
 
-  const [ showTradedInstrument, setShowTradedAsset ] = useState(false);
-  const [ node, setNode ] = useState<ClaimTreeNode | undefined>();
-
   const { getParty } = useParties();
   const ledger = useLedger();
   const party = useParty();
-  const { contracts: services, loading: l1 } = useStreamQueries(Service);
-  const { contracts: autoServices, loading: l2 } = useStreamQueries(AutoService);
-  const { contracts: instruments, loading: l3 } = useStreamQueries(Instrument);
-  const { contracts: derivatives, loading: l4 } = useStreamQueries(Derivative);
+  const inst = useInstruments();
 
-  const myServices = services.filter(s => s.payload.customer === party);
-  const myAutoServices = autoServices.filter(s => s.payload.customer === party);
-  const tradedInstrument = derivatives.find(c => c.payload.id.label === tradedInstrumentLabel);
-  const quotedInstrument = instruments.find(c => c.payload.id.label === quotedInstrumentLabel);
+  const svc = useServices();
+  const myListingServices = svc.listing.filter(s => s.payload.customer === party);
+  const myAutoListingServices = svc.listingAuto.filter(s => s.payload.customer === party);
 
-  const canRequest = !!tradedInstrumentLabel && !!tradedInstrument && !!quotedInstrumentLabel && !!quotedInstrument && !!id;
+  const tradableInstruments =
+    inst.generics.map(createInstrumentKey)
```

Comment:
I didn't use `inst.all` as I don't want for tokens to show up in the listing menu.

I think this is another use-case for instrument classification.

Reply:
Yes, true. We can consider adding some support for classification to instruments